rust/otap-dataflow/crates/otap/src/filter_processor/filter_logs.rs

```diff
@@ -0,0 +1,381 @@
+
+impl LogFilter {
+	pub fn filter(&self, logs_payload: OtapArrowRecord) -> Result<OtapArrowRecords, Error>{
+		let (include_resource_attr_filter, include_log_record_filter, include_log_attr_filter) = self.include.create_filters(logs_payload, false)?;
+		let (exclude_resource_attr_filter, exclude_log_record_filter, exclude_log_attr_filter) = self.exclude.create_filters(logs_payload, true)?;
+
+		// combine the include and exclude filters
+		let resource_attr_filter = arrow::compute::or_kleene(&include_resource_attr_filter, &exclude_resource_attr_filter).map_err()?;
+		let log_record_filter = arrow::compute::or_kleene(&include_log_record_filter, &exclude_log_record_filter).map_err()?;
+		let log_attr_filter = arrow::compute::or_kleene(&include_log_attr_filter, &exclude_log_attr_filter).map_err()?;
+
+		let resource_attrs = logs_payload.get(ArrowPayLoadType::ResourceAttrs)?;
+		let log_records = logs_payload.get(ArrowPayLoadType::Logs)?;
+		let log_attrs = logs_payload.get(ArrowPayLoadType::LogAttrs)?;
+
+
+        // HERE WE CLEAN UP THE TABLE WE EXTRACT THE IDS OF RECORDS WE ARE REMOVING AND MAKE SURE TO SYNC THESE REMOVALS WITH THE OTHER TABLES
+
+        // we start with the resource_attr we get the id of the resource_attr that are being removed and use that to update the log_records filter
+        // we want to remove the log_records the matching parent_id as the resource_attr that have been removed
+
+
+        // create filters that we will use to get the rows that are getting removed
+        // use these filters to get the ids of rows that are getting removed from the resource_attr
+        let id_resource_attr_filter = arrow::compute::not(&resource_attr_filter).map_err()?;
+        let resource_id_column = resource_attrs.column_by_name("parent_id").map_err()?;
+        // we have a have a array contain all the parent_ids that will be removed
+
+        // get ids that we being kept
+        let resource_ids_filtered = arrow::compute::filter(&resource_id_column, resource_attr_filter).map_err()?;
+        // get ids being removed
+        let resource_ids_removed = arrow::compute::filter(&resource_id_column, id_resource_attr_filter).map_err()?;
+
+        // update the log_records_filter to remove records that contain the parent_ids we found above
+        let log_record_resource_id_column = log_records.column_by_name("resource_id").map_err()?;
+        // init booleanarray here
+        let log_record_resouce_id_filter;
+
+
+        // TODO: we remove overlapping ids to only get the unique ids that are being removed, 
+        //we can move this logic to the get...filter functions to make sure the ids all stick together 
+        //that is parent_id with 0 should not both get removed or kept should be xor. In the get...filter functions
+        // after we get a filter that extracts all the attributes we should get the ids and use that to build our final filter
+        let resource_ids = resource_ids_removed - resource_ids_filtered;
```

Comment:
I'm not sure I understand the logic here. 

`resource_ids` is intended to be the unique IDs of resources we want to keep right?

But above, we compute:
```rs
        // get ids that we being kept
        let resource_ids_filtered = arrow::compute::filter(&resource_id_column, resource_attr_filter).map_err()?;
        // get ids being removed
        let resource_ids_removed = arrow::compute::filter(&resource_id_column, id_resource_attr_filter).map_err()?;
```

Isn't `resource_ids_filtered` the complement of `resource_ids_removed`?

I think what we'd really want here is the unique values in `resource_ids_filtered`, right?